apps/src/netsim/NetSimTable.js

```diff
@@ -68,51 +68,75 @@ var NetSimTable = module.exports = function (storageTable) {
   this.pollingInterval_ = DEFAULT_POLLING_DELAY_MS;
 };
 
+/**
+ * @param {!NodeStyleCallback} callback
+ */
 NetSimTable.prototype.readAll = function (callback) {
-  this.remoteTable_.readAll(function (data) {
-    callback(data);
-    if (data !== null) {
+  this.remoteTable_.readAll(function (err, data) {
+    callback(err, data);
+    if (err === null) {
       this.fullCacheUpdate_(data);
     }
   }.bind(this));
 };
 
+/**
+ * @param {!number} id
+ * @param {!NodeStyleCallback} callback
+ */
 NetSimTable.prototype.read = function (id, callback) {
-  this.remoteTable_.read(id, function (data) {
-    callback(data);
-    if (data !== undefined) {
+  this.remoteTable_.read(id, function (err, data) {
+    callback(err, data);
+    if (err === null) {
       this.updateCacheRow_(id, data);
```

Comment:
Not sure how much it ends up mattering here, but it feels a little bit weird to callback and then do something. Usually I would expect us to do our own processing, then send it to the callback as the last thing we do.